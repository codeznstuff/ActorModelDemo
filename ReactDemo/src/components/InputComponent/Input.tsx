import React from 'react';
import styles from './Input.module.scss';
import { inputProps } from '../../@types/constants';

export const Input = (props: inputProps) => (
  <table className={styles.input} data-testid="input">
    <tbody>
      <tr>
        <td id="previous" data-testid="previous">{props.previousInput}</td>
      </tr>
      <tr>
        <td data-testid="current">{props.input}</td>
      </tr>
    </tbody>
  </table>
)

export default Input;