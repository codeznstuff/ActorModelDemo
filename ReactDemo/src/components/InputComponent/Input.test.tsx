import React from 'react';
import { cleanup } from '@testing-library/react'
import { Input } from './Input';
import Enzyme, { shallow, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

describe("InputComponent", () => {
    let input: ShallowWrapper;
    beforeEach(() => {
        input = shallow(<Input previousInput={"PREVIOUS"} input={"CURRENT"} />);
    });
    it('Input Snapshot should match', () => {
        expect(input.debug()).toMatchSnapshot();
    })
    it('Input should have previous text equal to props', () => {
        expect(input.find('[data-testid="previous"]').text()).toBe("PREVIOUS")
    });
    it('Input should have current text equal to props', () => {
        expect(input.find('[data-testid="current"]').text()).toBe("CURRENT")
    });

    afterEach(cleanup);
})

