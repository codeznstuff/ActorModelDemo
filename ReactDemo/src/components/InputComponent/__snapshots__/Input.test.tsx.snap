// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputComponent Input Snapshot should match 1`] = `
"<table className=\\"input\\" data-testid=\\"input\\">
  <tbody>
    <tr>
      <td id=\\"previous\\" data-testid=\\"previous\\">
        PREVIOUS
      </td>
    </tr>
    <tr>
      <td data-testid=\\"current\\">
        CURRENT
      </td>
    </tr>
  </tbody>
</table>"
`;
